gradoop-flink/src/test/java/org/gradoop/flink/algorithms/gelly/clusteringcoefficient/GellyGlobalClusteringCoefficientDirectedTest.java

```diff
@@ -0,0 +1,93 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.flink.algorithms.gelly.clusteringcoefficient;
+
+import org.apache.commons.math3.util.Precision;
+import org.gradoop.common.model.impl.pojo.GraphHead;
+import org.gradoop.flink.model.api.epgm.LogicalGraph;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Test-class for {@link GellyGlobalClusteringCoefficientDirected}
+ */
+public class GellyGlobalClusteringCoefficientDirectedTest
+  extends GellyClusteringCoefficientTestBase {
+
+  /**
+   * Creates an instance of GellyGlobalClusteringCoefficientDirectedTest.
+   * Calls constructor of super class.
+   */
+  public GellyGlobalClusteringCoefficientDirectedTest() {
+    super();
+  }
+
+  @Override
+  public ClusteringCoefficientBase getCCAlgorithm() {
+    return new GellyGlobalClusteringCoefficientDirected();
+  }
+
+  @Override
+  public void testFullyConnectedGraph(LogicalGraph graph) throws Exception {
+    GraphHead head = graph.getGraphHead().collect().get(0);
+    assertEquals("Wrong global value for fully connected graph, should be 1", 1d,
+      head.getPropertyValue(ClusteringCoefficientBase.PROPERTY_KEY_GLOBAL).getDouble(), 0.0);
+  }
+
+  @Override
+  public void testNonConnectedGraph(LogicalGraph graph) throws Exception {
+    GraphHead head = graph.getGraphHead().collect().get(0);
+    double global = head.getPropertyValue(ClusteringCoefficientBase.PROPERTY_KEY_GLOBAL).getDouble();
+    assertTrue("Wrong global value for not connected graph, should be 0 or NaN",
+      Double.isNaN(global));
+  }
+
+  @Override
+  public void testSpecific() throws Exception {
+    String graphString = "halfConnected[" +
+      "/* half connected graph */" +
+      "(v0 {id:0, value:\"A\"})" +
+      "(v1 {id:1, value:\"B\"})" +
+      "(v2 {id:2, value:\"C\"})" +
+      "(v3 {id:3, value:\"D\"})" +
+      "(v0)-[e0]->(v1)" +
+      "(v0)-[e1]->(v2)" +
+      "(v0)-[e2]->(v3)" +
+      "(v1)-[e3]->(v2)" +
+      "]";
+
+    LogicalGraph graph = getLoaderFromString(graphString).getLogicalGraphByVariable("halfConnected");
+    LogicalGraph result = graph.callForGraph(getCCAlgorithm());
+
+    validateGraphProperties(result);
+
+    GraphHead head = result.getGraphHead().collect().get(0);
+
+    double global = head.getPropertyValue(ClusteringCoefficientBase.PROPERTY_KEY_GLOBAL)
+      .getDouble();
+    assertEquals("graph has wrong global value, should be 0.6",
+      Precision.round(3d / 5d, 3),
```

Comment:
Instead of round your result at this point you can use the 3th parameter in the `assertEquals` function to precise the scope of equality.

Reply:
Removed the Precision-usage..wasn't even necessary here - 3/5 = 0.6